cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -155,12 +155,14 @@ type DNSAddOnType string
 
 const (
 	// CoreDNS add-on type
+	// TODO: remove with v1beta2 https://github.com/kubernetes/kubeadm/issues/2459
 	CoreDNS DNSAddOnType = "CoreDNS"
 )
 
 // DNS defines the DNS addon that should be used in the cluster
 type DNS struct {
 	// Type defines the DNS add-on to be used
+	// TODO: remove with v1beta2 https://github.com/kubernetes/kubeadm/issues/2459
```

Comment:
Should we remove the field from internal types as well?
AFAIK the code for generating kube-dns manifest has been already removed, so we don't really need this anymore. Am I wrong?

Reply:
i'd happily remove it, but i don't think it would be possible to round trip a v1beta2 -> internal -> v1beta2 without this field.

Reply:
if the problem are roundtrips you can force the fuzzer to set DNS.Type to empty string, so the roundtrip  should pass

Reply:
discussed the problem with @fabriziopandini offline.
created a related PR here: https://github.com/kubernetes/kubernetes/pull/101598

i will update this one too, soon.